components/BarcodeScanner.vue

```diff
@@ -138,28 +155,51 @@ export default {
       cameraConstraints.zoom.step = step;
     }
 
+    async function emitScanResult({barcodeDetector, video}) {
+      const {signal} = abortController;
+      const barcodes = await detectBarcodes({barcodeDetector, video, signal});
+      const [result] = barcodes;
+      if(result?.format && result?.rawValue) {
+        emit('result', {type: result.format, text: result.rawValue});
+      }
+    }
+
+    function startBarcodeDetection() {
+      // Get video element
+      const video = document.querySelector(
+        '#dce-video-container > video'
+      );
+      const {BarcodeDetector} = globalThis;
+      // Check if BarcodeDetector is supported
+      if(!BarcodeDetector) {
+        alert('Barcode Detector is not supported in this browser.');
+        return;
+      }
+      const barcodeDetector = new BarcodeDetector({
+        formats: formatsToSupport
+      });
+        // Start the detection loop
+      video.requestVideoFrameCallback(
+        () => emitScanResult({barcodeDetector, video})
+      );
```

Comment:
This scheduling isn't needed anymore. We can just call `emitScanResult()`, all the video scheduling stuff is now handled:

```suggestion
      // emit the first scanned result
      emitScanResult({barcodeDetector, video});
```

Reply:
Addressed here 9497a7d